sdk-cpp/include/do_exceptions.h

```diff
@@ -1,21 +1,45 @@
 #ifndef _DELIVERY_OPTIMIZATION_DO_EXCEPTIONS_H
 #define _DELIVERY_OPTIMIZATION_DO_EXCEPTIONS_H
 
-#include <exception>
 #include <stdint.h>
 #include <system_error>
 
-#if defined(DO_INTERFACE_COM)
-#include <winerror.h>   // FAILED macro
+#if (!DO_DISABLE_EXCEPTIONS)
+#include <exception>
 #endif
 
 namespace microsoft
 {
 namespace deliveryoptimization
 {
 
+//These error macros are defined behind guards so that they do not redefine the macro for for platforms/toolchains in which they already exist
+#ifndef RETURN_RES_IF_FAILED
+#define RETURN_RES_IF_FAILED(res)  {  \
+    int32_t __res = (res);            \
+    if(FAILED(__res)) return __res;   \
+}
+#endif
+
+#ifndef FACILITY_DELIVERY_OPTIMIZATION 
+#define FACILITY_DELIVERY_OPTIMIZATION   208
+#endif
+
+#define DO_ERROR_FROM_SYSTEM_ERROR(x) (int32_t)(0xC0000000 | (0xD0 << 16) | ((int32_t)(x) & 0x0000FFFF))
+#define DO_ERROR_FROM_STD_ERROR(x) ((int32_t)(x) <= 0 ? ((int32_t)(x)) : ((int32_t) (((int32_t)(x) & 0x0000FFFF) | (FACILITY_DELIVERY_OPTIMIZATION << 16) | 0x80000000)))
+
+#ifndef FAILED
+#define FAILED(res) (((int32_t)(res)) < 0)
+#endif
+
+#ifndef SUCCEEDED
+#define SUCCEEDED(res) (((int32_t)(res)) >= 0)
+#endif
```

Comment:
I used RES here, but I think I could also replace it with 'e' for brevity, as well as all int32_t locals